### PR TITLE
MQE-1644: Add ability to see JS log in Allure 

### PR DESCRIPTION
**Pull request by KevinBKozan**
### Description
- Refactored ErrorLogger (no longer singleton, renamed for its actual use
- Added `stepAfter` actions to log browser logs

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -277,6 +277,24 @@ Example:
...
+
+Enables addition of browser logs to Allure steps
+
+```conf
+ENABLE_BROWSER_LOG=true
+```
+
+### BROWSER_LOG_BLACKLIST
+
+Blacklists types of browser log entries from appearing in Allure steps.
```
Comment (jilu1) [inline docs/configuration.md]: Can we add all possible types in the blacklist? are `other` and `console-api` the only possible types?

File: etc/config/.env.example [hunk 2]
```diff
@@ -54,4 +54,9 @@ MODULE_WHITELIST=Magento_Framework,ConfigurableProductWishlist,ConfigurableProdu
 
 #*** Default timeout for wait actions
 #WAIT_TIMEOUT=10
+
+#*** Uncomment and set to enable browser log entries on actions in Allure. Blacklist is used to filter logs of a specific "source"
+#ENABLE_BROWSER_LOG=true
+#BROWSER_LOG_BLACKLIST=other
```
Comment (jilu1) [inline etc/config/.env.example]: `console-api` is not in the blacklist, is this on purpose?

File: src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    /**
+     * Loops through given log and filters entries of the given type.
+     *
+     * @param array  $log
+     * @param string $type
+     * @return array
+     */
+    public static function filterLogsOfType($log, $type)
```
Comment (jilu1) [inline src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php]: `filterLogsOfType()` and `getLogsOfType` are complete the same. Any reason we need two functions here?

File: src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php [hunk 4]
```diff
@@ -173,7 +174,17 @@ public function beforeStep(\Codeception\Event\StepEvent $e)
      */
     public function afterStep(\Codeception\Event\StepEvent $e)
     {
-        ErrorLogger::getInstance()->logErrors($this->getDriver(), $e);
+        if (getenv('ENABLE_BROWSER_LOG')) {
+            $browserLog = $this->getDriver()->webDriver->manage()->getLog("browser");
+            foreach (explode(',', getenv('BROWSER_LOG_BLACKLIST')) as $source) {
+                $browserLog = BrowserLogUtil::filterLogsOfType($browserLog, $source);
+            }
```
Comment (jilu1) [inline src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php]: Can line 177 ~ 181 be moved to BrowserLogUtil class? The same logic is in BrowserLogUtil.

File: docs/configuration.md (hunk 1 above)
Comment (KevinBKozan) [reply docs/configuration.md]: I'm not 100% sure what all of the types are, I've ony been able to trigger those two.

If you can find documentation of the available types then please let me know, I've had no luck.

File: etc/config/.env.example (hunk 2 above)
Comment (KevinBKozan) [reply etc/config/.env.example]: Kind of, yes. `other` maps to miscelaneous Chrome specific warnings as far as I can tell, which wouldn't be particularly useful.

Do you think thi sblacklist should be blak by default?

File: src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php (hunk 3 above)
Comment (KevinBKozan) [reply src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php]: They are almost completely the same, check ln50 vs ln68

File: src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php (hunk 4 above)
Comment (KevinBKozan) [reply src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php]: Didn't watch this, I'll see if I can move it

Comment (KevinBKozan) [reply src/Magento/FunctionalTestingFramework/Extension/TestContextExtension.php]: I did it the other way around, I removed redundant logic from `BrowserLogUtil::logErrors()`. This kind of log getting should be done by the consumer of the Util, not by the util itself in my opinion.

File: src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php (hunk 3 above)
Comment (jilu1) [reply src/Magento/FunctionalTestingFramework/Extension/BrowserLogUtil.php]: Ok, I see the difference.

File: docs/configuration.md (hunk 1 above)
Comment (jilu1) [reply docs/configuration.md]: I see. Then let's not mention.

File: etc/config/.env.example (hunk 2 above)
Comment (jilu1) [reply etc/config/.env.example]: If it's on purpose, I am fine. No need to change.

**Pull request merged**